exercises/exercise-05/index.ts

```diff
@@ -78,6 +85,11 @@ function filterPersons(persons: Person[], personType: string, criteria: unknown)
 let usersOfAge23: User[] = filterPersons(persons, 'user', { age: 23 });
 let adminsOfAge23: Admin[] = filterPersons(persons, 'admin', { age: 23 });
 
+// @ts-ignore
+let astronautAdmins: Admin[] = filterPersons(persons, 'admin', { occupation: 'Astronaut' });
+
+let peopleOfAge23: Person[] = filterPersons(persons, Math.random() < 0.5 ? 'user' : 'admin', {age: 23});
```

Comment:
This call results in an error with the overloaded definitions, even though it is guaranteed to be valid. This is one of the problems with overloads, and it's a reason to prefer using conditional types. Rewriting this function to use conditional types instead of an overload is much harder but might be a good exercise for later.

You can learn more about why conditional types are preferred in [this fantastic artsy post](https://artsy.github.io/blog/2018/11/21/conditional-types-in-typescript/) or Item 50 (Prefer Conditional Types to Overloaded Declarations) in _Effective TypeScript_.